### ISSUE TITLE
panos_commit_push not working

### ISSUE THREAD

**Issue by shen3lu4** [PaloAltoNetworks/pan-os-ansible#178]
## Describe the bug

panos_commit_push throwing errors

## Expected behavior

I am trying to use panos_commit_push module for the following:
`  - panos_commit_push:
      provider: '{{ provider }}'
      style: 'template stack'
      name: 'template-stack-name'`
(the above code should be on different lines)

Expect a job created for the push.

## Current behavior

It is returning the following error and job was not created:
 
The full traceback is:
Traceback (most recent call last):
  File "/home/cloud-user/.ansible/tmp/ansible-tmp-1611178289.8068113-25261-267685302036271/AnsiballZ_panos_commit_push.py", line 102, in <module>
    _ansiballz_main()
  File "/home/cloud-user/.ansible/tmp/ansible-tmp-1611178289.8068113-25261-267685302036271/AnsiballZ_panos_commit_push.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/cloud-user/.ansible/tmp/ansible-tmp-1611178289.8068113-25261-267685302036271/AnsiballZ_panos_commit_push.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.paloaltonetworks.panos.plugins.modules.panos_commit_push', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib64/python3.6/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_panos_commit_push_payload_g2__pwvs/ansible_panos_commit_push_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_commit_push.py", line 201, in <module>
  File "/tmp/ansible_panos_commit_push_payload_g2__pwvs/ansible_panos_commit_push_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_commit_push.py", line 183, in main
  File "/usr/local/lib/python3.6/site-packages/panos/base.py", line 4482, in commit
    sync=sync, exception=exception, cmd=cmd, admins=admins, sync_all=sync_all
  File "/usr/local/lib/python3.6/site-packages/panos/base.py", line 4583, in _commit
    result = self.syncjob(commit_response, sync_all=sync_all)
  File "/usr/local/lib/python3.6/site-packages/panos/base.py", line 4690, in syncjob
    return self._parse_job_results(job_xml, get_devices=True)
  File "/usr/local/lib/python3.6/site-packages/panos/base.py", line 4854, in _parse_job_results
    "warnings": job["warnings"],
KeyError: 'warnings'
fatal: [ado_lab_pano]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/cloud-user/.ansible/tmp/ansible-tmp-1611178289.8068113-25261-267685302036271/AnsiballZ_panos_commit_push.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/home/cloud-user/.ansible/tmp/ansible-tmp-1611178289.8068113-25261-267685302036271/AnsiballZ_panos_commit_push.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/cloud-user/.ansible/tmp/ansible-tmp-1611178289.8068113-25261-267685302036271/AnsiballZ_panos_commit_push.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.paloaltonetworks.panos.plugins.modules.panos_commit_push', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_panos_commit_push_payload_g2__pwvs/ansible_panos_commit_push_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_commit_push.py\", line 201, in <module>\n  File \"/tmp/ansible_panos_commit_push_payload_g2__pwvs/ansible_panos_commit_push_payload.zip/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_commit_push.py\", line 183, in main\n  File \"/usr/local/lib/python3.6/site-packages/panos/base.py\", line 4482, in commit\n    sync=sync, exception=exception, cmd=cmd, admins=admins, sync_all=sync_all\n  File \"/usr/local/lib/python3.6/site-packages/panos/base.py\", line 4583, in _commit\n    result = self.syncjob(commit_response, sync_all=sync_all)\n  File \"/usr/local/lib/python3.6/site-packages/panos/base.py\", line 4690, in syncjob\n    return self._parse_job_results(job_xml, get_devices=True)\n  File \"/usr/local/lib/python3.6/site-packages/panos/base.py\", line 4854, in _parse_job_results\n    \"warnings\": job[\"warnings\"],\nKeyError: 'warnings'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

## Steps to reproduce

- create a template in panorama manually named 'test-template-stack'
- create a playbook that contains this task (provider is the credential to the pano)

  `- panos_commit_push: 
      provider: '{{ provider }}'
      style: 'template stack'
      name: 'test-template-stack'`

- please note the above code block should be on different lines
- run the playbook

## Context

This affects my ability to automate the push of a configuration in pano. 

## Your Environment

- Collection: 2.4.0
- Python: 3.6.8
- Ansible: 2.10.4
- PAN-OS Python Library & version (e.g. pandevice 0.14.0, pan-os-python 1.0.2): pan-os-python (1.0.2) & pan-python (0.16.0)

**Comment by mrichardson03** [comment]
Can you provide steps to reproduce?  This appears to be a bug in the upstream library [pan-os-python](https://github.com/PaloAltoNetworks/pan-os-python), and an issue will have to be opened there.

**Comment by shen3lu4** [comment]
Added steps to reproduce above.
Shall I open an issue in pan-os-python or you will do that?

**Comment by JulioPDX** [comment]
panos_commit_push not working for me as well. In my case trying to push using `style:  'log collector group'`

**Comment by shinmog** [comment]
@shen3lu4 @JulioPDX 

Could either of you please provide the returned XML from querying the job status of a log collector commit-all?  That error message is kind of implying that the job status has a completely different format from what is expected.

You can do this by doing the following API call:

```python
from panos.panorama import Panorama

# Use your real credentials here instead.
pano = Panorama(host, user, passwd)

# Replace 1234 with the actual job ID.
print(pano.op('show jobs id "1234"', xml=True))
```

**Comment by jamesholland-uk** [comment]
I've hit the same issue, same error message put as tom0010 above when using the commit_push module. I tried working around it by attempting to use panos_op and panos_type_cmd to execute the commit-push, but they both failed with illegal cmd:
![Screenshot 2021-04-21 at 00 10 37](https://user-images.githubusercontent.com/6574404/117853646-92992480-b280-11eb-8838-16258fb35a4d.png)
![Screenshot 2021-04-21 at 00 10 56](https://user-images.githubusercontent.com/6574404/117853650-93ca5180-b280-11eb-9ebc-7fd227cb4a1b.png)

**Comment by mrichardson03** [comment]
> I've hit the same issue, same error message put as tom0010 above when using the commit_push module. I tried working around it by attempting to use panos_op and panos_type_cmd to execute the commit-push, but they both failed with illegal cmd:
> ![Screenshot 2021-04-21 at 00 10 37](https://user-images.githubusercontent.com/6574404/117853646-92992480-b280-11eb-8838-16258fb35a4d.png)
> ![Screenshot 2021-04-21 at 00 10 56](https://user-images.githubusercontent.com/6574404/117853650-93ca5180-b280-11eb-9ebc-7fd227cb4a1b.png)

That won't work, unfortunately.  `panos_type_cmd` would actually have to add a `type` parameter, because a `commit-all` actually looks like this:

```
https://{{host}}/api/?type=commit&action=all&cmd=<commit-all><shared-policy><include-template>no</include-template><device-group><entry name="norcal"/></device-group></shared-policy></commit-all>
```

**Comment by jamesholland-uk** [comment]
> That won't work, unfortunately. `panos_type_cmd` would actually have to add a `type` parameter, because a `commit-all` actually looks like this:
> ```
> https://{{host}}/api/?type=commit&action=all&cmd=<commit-all><shared-policy><include-template>no</include-template><device-group><entry name="norcal"/></device-group></shared-policy></commit-all>
> ```
Apologies, my bad! Workaround failed :-)

**Comment by btorresgil** [comment]
Fixed in pan-os-python v1.1.1.
